### ISSUE TITLE
[FR] Add a "description" column in the main panel

### ISSUE THREAD

**Issue by fzyzcjy** [mlflow/mlflow#11503]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

Hi thanks for the helpful tool! I would appreciate it if there could be one more row - (run) description - in this page. Then it is easy to glance through all runs with descriptions. IMHO this may not be super hard to implement, 

![image](https://github.com/mlflow/mlflow/assets/5236035/b76bd614-2090-4154-96f1-40648f52882f)

### Motivation

> #### What is the use case for this feature?
(see above)

> #### Why is this use case valuable to support for MLflow users in general?
(see above)

> #### Why is this use case valuable to support for your project(s) or organization?
(see above)

> #### Why is it currently difficult to achieve this use case?
(see above)

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
> if there could be one more row - (run) description - in this page. Then it is easy to glance through all runs with descriptions.

Could you elaborate it ? What's the content in the description ? do you have an example ? Thanks!

**Comment by fzyzcjy** [comment]
@WeichenXu123 Hi, IMHO mlflow has a good feature that we can add descriptions to a run:

![image](https://github.com/mlflow/mlflow/assets/5236035/4ca08db3-8851-4090-adf2-95c4ea8caabc)

This is the "description" I mean. For different people it can be different thing, and I guess you (mlflow devs) must know much more scenarios to use it than I know! As for my own case, I want to write down a sentence to remind me of this run's special thing, e.g. "fixed bug xxx", "let's try blahblah thing".

**Comment by WeichenXu123** [comment]
oh got it, thank you.

Then I think you mean to add a 'description' column (not row ) in the table to show the description of the Run ?

**Comment by fzyzcjy** [comment]
Yes you are right. I said that in the title and made a silly typo!

**Comment by WeichenXu123** [comment]
Make sense to this. I will talk with our team folks later for this FR.

**Comment by fzyzcjy** [comment]
Looking forward to it!

**Comment by WeichenXu123** [comment]
@hubertzub-db What do you think of this feature request ?

**Comment by hubertzub-db** [comment]
@fzyzcjy thanks for filing this FR! I think that if there's a use case for that, we definitely can add the description column. However what concerns me is not the adding the column by itself, but the approach to render its content. Run descriptions can contain a lot of rich content (markdown) and multi-line text. If we were to attempt displaying it as on the run details page, we might inadvertently hinder table readability. Maybe we could just display `n` few first words with no line breaks and use ellipsis (`...`) at the end? 🤔  Does it apply to your use case @fzyzcjy ?
CC @WeichenXu123 also @AbeOmor

**Comment by fzyzcjy** [comment]
> Maybe we could just display n few first words with no line breaks and use ellipsis (...) at the end?

Yes I think so! And maybe just render as pure text. For example, suppose someone has `## This is title\n\nThis is content`, we may just render as something like `This is title This is content`, instead of the huge title font.

**Comment by hubertzub-db** [comment]
@fzyzcjy great! Let us know if you're willing to provide even a minimal implementation (unformatted text in the column hidden by default) and if you need any guidelines for that :)

**Comment by fzyzcjy** [comment]
@hubertzub-db Hi, I am interested in doing so, but may not have enough time (e.g. I need to squeeze time to maintain my open source lib https://github.com/fzyzcjy/flutter_rust_bridge) at least recently.

**Comment by TejjeT** [comment]
Hello ,
I am interested in working on this issue. Could you assign this to me.

**Comment by hubertzub-db** [comment]
@TejjeT sure. Let me know if you need any guidelines for that.

**Comment by zhouyou9505** [comment]
Hello sir, Has this project started? If not, I am also interested. @hubertzub-db

**Comment by hubertzub-db** [comment]
@zhouyou9505 To date we didn't get any updates from @TejjeT nor @fzyzcjy so it's still probably fine to pick it up!

**Comment by fzyzcjy** [comment]
Yes, I will probably have no time for this, and looking forward to your PR!

**Comment by zhouyou9505** [comment]
> Yes, I will probably have no time for this, and looking forward to your PR!

@hubertzub-db Ok. could you reassign it to me

**Comment by zhouyou9505** [comment]
@hubertzub-db hi ,sir ,Please check the feature  #11996

**Comment by harupy** [comment]
Closed by #11996
